packages/@aws-cdk/pipelines/lib/blueprint/step-output.ts

```diff
@@ -0,0 +1,109 @@
+import { IResolvable, IResolveContext, Token, Tokenization } from '@aws-cdk/core';
+import { Step } from './step';
+
+const STEP_OUTPUT_SYM = Symbol.for('@aws-cdk/pipelines.StepOutput');
+
+/**
+ * A symbolic reference to a value produced by another step
+ *
+ * The engine is responsible for defining how these should be rendered.
+ */
+export class StepOutput implements IResolvable {
+  /**
+   * Return true if the given IResolvable is a StepOutput
+   */
+  public static isStepOutput(resolvable: IResolvable): resolvable is StepOutput {
+    return !!(resolvable as any)[STEP_OUTPUT_SYM];
+  }
+
+  /**
+   * Find all StepOutputs referenced in the given structure
+   */
+  public static findAll(structure: any): StepOutput[] {
+    return findAllStepOutputs(structure);
+  }
+
+  /**
+   * The step that produces this output
+   */
+  public readonly step: Step;
+
+  /**
+   * Additional data on the output, to be interpreted by the engine
+   */
+  public readonly engineSpecificInformation: any;
+
+  public readonly creationStack: string[] = [];
```

Comment:
doesn't this need a docustring?

Reply:
The docstring is inherited from the interface that declares this initially.